lib/rust/ensogl/core/src/display/world.rs

```diff
@@ -76,33 +76,35 @@ impl World {
     pub fn new(dom: &web_sys::HtmlElement) -> World {
         let logger = Logger::new("world");
         let stats = default();
+        let performance = web::performance();
         let scene_dirty = dirty::SharedBool::new(Logger::new_sub(&logger, "scene_dirty"), ());
         let on_change = enclose!((scene_dirty) move || scene_dirty.set());
         let scene = Scene::new(dom, &logger, &stats, on_change);
         let uniforms = Uniforms::new(&scene.variables);
         let main_loop = animation::DynamicLoop::new();
-        let stats_monitor = stats::Monitor::new(&stats);
+        let stats_monitor = debug::monitor::Monitor::new();
         let on_before_frame = <callback::SharedRegistryMut1<animation::TimeInfo>>::new();
         let on_after_frame = <callback::SharedRegistryMut1<animation::TimeInfo>>::new();
         let on_stats_available = <callback::SharedRegistryMut1<StatsData>>::new();
+        let stats_draw_handle = on_stats_available.add(f!([stats_monitor] (stats: &StatsData) {
+            stats_monitor.draw(stats);
+        }));
         let main_loop_frame = main_loop.on_frame(
-            f!([stats_monitor,on_before_frame,on_after_frame,on_stats_available,uniforms,scene_dirty,scene]
+            f!([stats,on_before_frame,on_after_frame,on_stats_available,uniforms,scene_dirty,scene]
             (t:animation::TimeInfo) {
                 // Note [Main Loop Performance]
 
-                stats_monitor.begin();
+                let previous_frame_stats = stats.begin_frame(performance.now());
```

Comment:
I think this inconsistency is because `begin_frame` is misplaced. It is an operation for building up a sequence of `StatsData` values; it seems odd for sequence logic to be in a method on a `StatsData` object.